src/logger.ts

```diff
@@ -72,7 +72,7 @@ export function logOnce(cacheKey: string, cb: () => void) {
 	}
 }
 
-export function initializeLogger(): void {
+export function initializeLogger(options: RuntimeConfig): void {
```

Comment:
I was operating under the assumption that nothing that _would_ need to be redacted will be logged until after validation completes and we set the runtime config. Is that the case? Now I'm wondering, maybe not, specifically in the event that there are errors in the config. What do you think? If my initial assumption was wrong then I think what you already wrote makes the most sense.

Reply:
Honestly, I haven't really tested against the torznab or client URL's which are the only things I'm aware of that need and have redaction....but I assume that if there was a problem with them (in this case if they do not match `.url()` in Zod then it would leak the API key or password that is being used.

Let me test the output of torznab and deluge's client URL and see what it looks like real quick, I'll followup in a few minutes.

Reply:
Just tested, neither output the actual URL, and torznab does similar to the dataDirs thing I described - which would require splitting it on the comma (it says torznab,0) - this could either be something we leave and I incorporate that same thing to dataDirs like you had previously mentioned, and parse it in the error message if comma's exist for the URL - or I'd have to alternatively write an entire function for each array that we parse like this. 

Because of this I think it would be better to just handle it in the error messaging of the doc's URL, it prevents us from having to handle this differently now, and if we add any arrays to the config in the future.

Either way, I don't really see how we could end up logging anything with sensitive information before setting runtimeconfig.

Reply:
Okay, I think your solution as-is is good then.